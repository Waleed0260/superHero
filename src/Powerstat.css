.ho{
    /* text-align: center; */
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 100px;
}
/* .power-stat{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
} */
.power{
    display: flex;
    flex-direction: column;
}
.img{
    display: flex;
    flex: 1;
    margin-top: 50px;
}
.img>img{
    height: 450px;
    /* width: 380px; */
    width: 28vw;
    /* margin-top: -70px; */
    margin-left: 30px;
}
.img>h1{
    margin-top: -10px;
    color: orange;
    margin-left: 20px;
    color: rgb(255, 255, 255);
   
}
.img>h1>span{
 color: orange;

}
.stats{
    display: flex;
    flex: 1;
    /* margin-left: -320px; */
    width: 50vw;
    position: absolute;
    right: 120px;
    /* background-color: blue; */
    margin-top: 100px;
}
.stats>ul>li{
    margin-top: 30px;
    list-style: none;
    display: flex;
    justify-content: space-between;
    gap: 29vw;
    border-bottom: 4px solid rgb(41, 41, 41);
}
.stats>ul>li>:nth-of-type(1){
    text-transform: uppercase;
}
.stats>ul>li>:nth-of-type(2){
    color: rgb(255, 166, 0);
}


/* 
@media screen and (max-width:900px) {
    .img>img{
        width: 75vw;
        display: flex;
        justify-content: center;
        align-items: center;
        margin-left: 90px;
        margin-top: 60px;
    }
    .img>h1{
        position: absolute;
        left: 34vw;
    }
    .stats{
        position: absolute;
        top: 40vh;
    }
} */

@media screen and (max-width:768px) {
    .stats{
        margin-top: 560px;
        /* right: 320px; */
        left: 40px;
        /* margin-right: 200px; */
    }
    .stats>ul>li{
        gap: 62vw;
    }

    .img>img{
        width: 70vw;
        margin-top: 60px;
        height: 360px;
        margin-left: 60px;
    }
    .img>h1{
        margin-top: 10px;
        margin-left: -300px;
    }
}


@media screen and (max-width:600px) {
 .stats>ul>li{
    gap: 47vw;
 }   
 .img>img{
    margin-left: 40px;
 }
 .img>h1{
    font-size: 200;
    margin-left:-350px;
 }
 .img{
    margin-top: 80px;
 }
 .stats{
    margin-top: 580px;
 }
}


@media screen and (max-width:500px) {
    .stats{
        left: -20px;
    }
    .stats>ul>li{
        gap: 58vw;
    }
}

@media screen and (max-width:460px) {
    .img>img{
        margin-left: 20px;
    }
}

@media screen and (max-width:410px) {
    .img>h1{
        margin-left: -320px;
    }
    .img>img{
        width: 78vw;
        margin-left: -10px;
    }
}


@media screen and (max-width:350px) {
    .img>h1{
        margin-left: -290px;
        font-weight: 500;
    }
}

