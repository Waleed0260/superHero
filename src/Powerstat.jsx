import React, { useEffect, useState } from "react";
// import { json } from 'react-router-dom'
import "./Powerstat.css";
const Powerstat = (props) => {
  
  const [result, setResult] = useState([]);
  const [name, setName] = useState([]);
  const [img, setImg] = useState([]);
  const [spinner, setSpinner] = useState("");

  // const fetchData = async () => {
  //   const data = await fetch(
  //     "https://www.superheroapi.com/api.php/771712793908121/search/batman/Batman"
  //   ).then((res) => res.json());
  //   setResult(data.results[0]);
  // };
  // useEffect(() => {
  //   fetchData();
  // }, []);
  const capitalizeFirstLetter = (string) => {
    return string.charAt(0).toUpperCase() + string.slice(1);
  };



  useEffect(() => {
    const fetchApi = async () => {
      setSpinner(true);
      props.setProgress(30)
      const url = `https://www.superheroapi.com/api.php/771712793908121/search/${props.name}`;
      const response = await fetch(url);
      const resJson = await response.json();
      props.setProgress(50);
      setResult(resJson.results[0].powerstats);
      setName(resJson.results[0]);
      setImg(resJson.results[0].image);
      setSpinner(false);
      props.setProgress(100)
      //  console.log(resJson.strength);
    };
    fetchApi();
  }, []);
  
  useEffect(()=>{
    document.title  = `${capitalizeFirstLetter(
      props.pov
    )} - SuperHero`
  }, []);




  return (
    <>
      {spinner ? (
        <div className="ho">
          <h1>LOADING...</h1>
        </div>
      ) : (
        <div className="power-stat">
          <>
            <div className="power" key={name.id}>
              <div className="img">
                <img src={img.url} alt="" />
                <h1>{name.name}-<span>{props.pov}</span></h1>
              </div>
              <div className="stats">
                <ul>
                  <li>
                    <span>Intelligance</span>
                    <span>{result.intelligence}</span>
                  </li>
                  <li>
                    <span>Speed</span>
                    <span>{result.speed}</span>
                  </li>
                  <li>
                    <span>Strength</span>
                    <span>{result.strength}</span>
                  </li>
                  <li>
                    <span>Durability</span>
                    <span>{result.durability}</span>
                  </li>
                  <li>
                    <span>Power</span>
                    <span>{result.power}</span>
                  </li>
                  <li>
                    <span>Combat</span>
                    <span>{result.combat}</span>
                  </li>
                </ul>
              </div>
            </div>
          </>
        </div>
      )}
      {/* <button data-testid="button">{label}</button> */}
    </>
  );
};

export default Powerstat;
