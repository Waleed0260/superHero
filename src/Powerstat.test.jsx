import React from "react";
import ReactDOM from "react-dom";
import Powerstat from "./Powerstat";
import renderer from "react-test-renderer"

// import {render} from 'react-dom';
import "@testing-library/jest-dom/extend-expect"

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Powerstat></Powerstat>, div)
})


it("matches snapshot",()=>{
    const tree = renderer.create(<Powerstat></Powerstat>).toJSON();
    expect(tree).toMatchSnapshot();
})