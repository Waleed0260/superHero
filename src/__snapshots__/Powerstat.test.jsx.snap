// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="power-stat"
>
  <div
    className="power"
  >
    <div
      className="img"
    >
      <img
        alt=""
      />
      <h1>
        -
        <span />
      </h1>
    </div>
    <div
      className="stats"
    >
      <ul>
        <li>
          <span>
            Intelligance
          </span>
          <span />
        </li>
        <li>
          <span>
            Speed
          </span>
          <span />
        </li>
        <li>
          <span>
            Strength
          </span>
          <span />
        </li>
        <li>
          <span>
            Durability
          </span>
          <span />
        </li>
        <li>
          <span>
            Power
          </span>
          <span />
        </li>
        <li>
          <span>
            Combat
          </span>
          <span />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
